pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/daemonset.go

```diff
@@ -49,8 +52,41 @@ func ExtractDaemonSet(ds *appsv1.DaemonSet) *model.DaemonSet {
 		daemonSet.Spec.Selectors = extractLabelSelector(ds.Spec.Selector)
 	}
 
+	if len(ds.Status.Conditions) > 0 {
+		dsConditions, conditionTags := extractDaemonSetConditions(ds)
+		daemonSet.Conditions = dsConditions
+		daemonSet.Tags = append(daemonSet.Tags, conditionTags...)
+	}
+
 	daemonSet.Spec.ResourceRequirements = ExtractPodTemplateResourceRequirements(ds.Spec.Template)
 	daemonSet.Tags = append(daemonSet.Tags, transformers.RetrieveUnifiedServiceTags(ds.ObjectMeta.Labels)...)
 
 	return &daemonSet
 }
+
+// extractDaemonSetConditions iterates over daemonset conditions and returns:
+// - the payload representation of those conditions
+// - the list of tags that will enable pod filtering by condition
+func extractDaemonSetConditions(p *appsv1.DaemonSet) ([]*model.DaemonSetCondition, []string) {
+	conditions := make([]*model.DaemonSetCondition, 0, len(p.Status.Conditions))
+	conditionTags := make([]string, 0, len(p.Status.Conditions))
+
+	for _, condition := range p.Status.Conditions {
+		c := &model.DaemonSetCondition{
+			Message: condition.Message,
+			Reason:  condition.Reason,
+			Status:  string(condition.Status),
+			Type:    string(condition.Type),
+		}
+		if !condition.LastTransitionTime.IsZero() {
+			c.LastTransitionTime = condition.LastTransitionTime.Unix()
+		}
+
+		conditions = append(conditions, c)
+
+		conditionTag := fmt.Sprintf("kube_condition_%s:%s", strings.ToLower(string(condition.Type)), strings.ToLower(string(condition.Status)))
```

Comment:
nit: this line could be a common function so the tag format is only declared once

Reply:
[d7f518b](https://github.com/DataDog/datadog-agent/pull/16499/commits/d7f518bfc5d0fc86e364d6596fca4d9999c12cb3)